extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/model/GreenbidsUserAgent.java

```diff
@@ -0,0 +1,56 @@
+package org.prebid.server.hooks.modules.greenbids.real.time.data.model;
+
+//import com.iab.openrtb.request.UserAgent;
+import ua_parser.Client;
+import ua_parser.Parser;
+import ua_parser.UserAgent;
+
+import java.util.Set;
+
+public class GreenbidsUserAgent {
+
+    public static final Set<String> PC_OS_FAMILIES = Set.of(
+            "Windows 95", "Windows 98", "Solaris");
+
+    private static final Parser UA_PARSER = new Parser();
+
+    private final String userAgentString;
+
+    private final UserAgent userAgent;
+
+    private final ua_parser.Device device;
+
+    private final ua_parser.OS os;
+
+    public GreenbidsUserAgent(String userAgentString) {
+        this.userAgentString = userAgentString;
```

Comment:
If `userAgentString` is null (or any other feature is null) we should still be able to do the model inference with what is available. Because probably if this is the case the model is also trained on the data with missing columns, so they should not be a problem.

Reply:
So you have to handle the null case in the following method:
```java
    public boolean isPC() {
        return userAgentString.contains("Windows NT")
                || PC_OS_FAMILIES.contains(os.family)
                || ("Windows".equals(os.family) && "ME".equals(os.major))
                || ("Mac OS X".equals(os.family) && !userAgentString.contains("Silk"))
                || userAgentString.contains("Linux") && userAgentString.contains("X11");
    }
```

Reply:
fixed, added null handling for user agent inside the `GreenbidsUserAgent` class